lib/authn.js

```diff
@@ -0,0 +1,90 @@
+// @ts-check
+const config = require('../lib/config');
+const csrf = require('../lib/csrf');
+const sqldb = require('@prairielearn/postgres');
+
+const sql = sqldb.loadSqlEquiv(__filename);
+
+/**
+ * @typedef {Object} LoadUserOptions
+ * @property {boolean} [LoadUserOptions.pl_authn_cookie=true] - Create the cookie?
+ * @property {boolean} [LoadUserOptions.redirect=false] - Redirect after processing?
+ * @property {boolean} [LoadUserOptions.locals_authn=true] - Populate res.locals.authn*?
+ */
+const optionsDefault = {
```

Comment:
Nit: this object is only used in one place; I'd be inclined to just inline these properties into the first line of `loadUser`.

Reply:
I prefer to give the library two objects keeping the purpose separate. The first has all of the user data. The second tells the library what actions to take with that data.

Reply:
Sorry, I should have been more explicit.

```diff
- let options = { ...optionsDefault, ...optionsParams };
+ let options = { pl_authn_cookie: true, redirect: false, ...optionsParams };
```